plugins/slack_app/test/controllers/slack_app_controller_test.rb

```diff
@@ -0,0 +1,100 @@
+require_relative '../test_helper'
+
+SingleCov.covered!
+
+def post_as(id, params)
+  opts = params.merge(
+    token: nil,
+    user_id: id.identifier,
+    response_url: 'http://example.com/blah'
+  )
+  post :command, opts
+end
+
+describe SlackAppController do
+  let(:user1) { users(:deployer) }
+  let(:user2) { users(:deployer_buddy) }
+  let(:user1_identifier) { slack_identifiers(:user1) }
+  let(:user2_identifier) { slack_identifiers(:user2) }
+  before do
+    user1_identifier.update_column(:user_id, user1.id)
+    user2_identifier.update_column(:user_id, user2.id)
+  end
+
+  as_a_viewer do
+    describe '#oauth' do
+      with_env SLACK_CLIENT_ID: 'client-id', SLACK_CLIENT_SECRET: 'client-secret' do
+        it 'sends the user to Slack to connect the app' do
+          get :oauth
+          assert_match 'https://slack.com/oauth/authorize?client_id=', @response.body
```

Comment:
assert_match produces nasty output when it fails ... include is much nicer there